src/utils/runtime/src/logger.rs

```diff
@@ -247,53 +247,73 @@ pub fn init_risingwave_logger(settings: LoggerSettings) {
                 e.as_report(),
             )
         });
-        let file = std::fs::OpenOptions::new()
-            .create(true)
-            .write(true)
-            .truncate(true)
-            .open(query_log_path.join("query.log"))
-            .unwrap_or_else(|e| {
-                panic!(
-                    "failed to create '{}/query.log': {}",
-                    query_log_path.display(),
-                    e.as_report(),
-                )
-            });
-        let layer = tracing_subscriber::fmt::layer()
-            .with_ansi(false)
-            .with_level(false)
-            .with_file(false)
-            .with_target(false)
-            .with_timer(default_timer.clone())
-            .with_thread_names(true)
-            .with_thread_ids(true)
-            .with_writer(std::sync::Mutex::new(file))
-            .with_filter(filter::Targets::new().with_target(PGWIRE_QUERY_LOG, Level::TRACE));
-        layers.push(layer.boxed());
 
-        let file = std::fs::OpenOptions::new()
-            .create(true)
-            .write(true)
-            .truncate(true)
-            .open(query_log_path.join("slow_query.log"))
-            .unwrap_or_else(|e| {
-                panic!(
-                    "failed to create '{}/slow_query.log': {}",
-                    query_log_path.display(),
-                    e.as_report(),
-                )
-            });
-        let layer = tracing_subscriber::fmt::layer()
-            .with_ansi(false)
-            .with_level(false)
-            .with_file(false)
-            .with_target(false)
-            .with_timer(default_timer)
-            .with_thread_names(true)
-            .with_thread_ids(true)
-            .with_writer(std::sync::Mutex::new(file))
-            .with_filter(filter::Targets::new().with_target(SLOW_QUERY_LOG, Level::TRACE));
-        layers.push(layer.boxed());
+        /// Newtype wrapper for `DefaultFields`.
+        ///
+        /// `fmt::Layer` will share the same `FormattedFields` extension for spans across
+        /// different layers, as long as the type of `N: FormatFields` is the same. This
+        /// will cause several problems:
+        ///
+        /// - `with_ansi(false)` does not take effect and it will follow the settings of
+        ///   the primary fmt layer installed above.
+        /// - `Span::record` will update the same `FormattedFields` multiple times,
+        ///   leading to duplicated fields.
+        ///
+        /// As a workaround, we use a newtype wrapper here to get a different type id.
+        /// The const generic parameter `SLOW` is further used to distinguish between the
+        /// query log and the slow query log.
+        #[derive(Default)]
+        struct FmtFields<const SLOW: bool>(DefaultFields);
```

Comment:
Where is the `SLOW` used?

Reply:
https://github.com/risingwavelabs/risingwave/blob/9dcad747a462f9dd1df571f8812ffa5f36596197/src/utils/runtime/src/logger.rs#L303-L306

There's no actual use for this flag. It just makes the compiler to treat them as different types and then assign them with different [`type_id`](https://doc.rust-lang.org/stable/std/any/trait.Any.html#tymethod.type_id)s, which is used for RTII.